omigami/ms2deep.py

```diff
@@ -0,0 +1,148 @@
+import ast
+import json
+from logging import getLogger
+from typing import Dict, Union, List, Generator
+
+import requests
+from matchms import Spectrum
+from matchms.importing import load_from_mgf
+
+
+Payload = Dict[str, Dict[str, Dict[str, Union[int, dict]]]]
+log = getLogger(__file__)
+
+JSON = Union[List[dict], dict]
+
+
+class InvalidCredentials(Exception):
+    pass
+
+
+class InvalidNumberOfSpectra(Exception):
+    pass
+
+
+class MS2Deep:
+    _endpoint_url = (
+        "https://omigami.datarevenue.com/seldon/seldon/ms2deep/api/v0.1/predictions"
+    )
+
+    def __init__(self, token: str):
+        self._token = token
+
+    def predict_similarity_of_pair(
+        self,
+        mgf_path: str,
+    ) -> float:
+        """
+        Predicts molecular structural similarities (Tanimoto scores) from pairs of
+        mass spectrometry spectra in local mgf files using ms2deepscore algorithm
+
+        Parameters
+        ----------
+        mgf_path: str
+            Local path to mgf file with two spectra
+
+        Returns
+        -------
+        Molecular structural similarities (Tanimoto scores)
+
+        """
+        spectra_generator = load_from_mgf(mgf_path)
+
+        payload = self._build_payload(spectra_generator)
+        response = self._send_request(payload)
+
+        prediction = self._format_results(response)
+        return prediction
+
+    def _build_payload(
+        self,
+        pair: Generator[Spectrum, None, None],
+    ) -> JSON:
+        """Build the json payload
+
+        Returns:
+        --------
+        - payload: JSON
+            the full request payload with input data
+        """
+        spectra = []
+        for spectrum in pair:
+            spectra.append(
+                {
+                    "peaks_json": str(
+                        [
+                            [mz, intensity]
+                            for mz, intensity in zip(
+                                spectrum.peaks.mz, spectrum.peaks.intensities
+                            )
+                        ]
+                    ),
+                }
+            )
+
+        self._validate_input(spectra)
+
+        payload = {
+            "data": {
+                "ndarray": {
+                    "parameters": [],  # maybe there will be some later on
```

Comment:
I think these are better off as curly braces `{}` since we will probably check for kwargs